apps/web/app/api/epub/route.ts

```diff
@@ -0,0 +1,65 @@
+import { inArray } from "drizzle-orm";
+import epub, { Chapter } from "epub-gen-memory";
+
+import { db } from "@hoarder/db";
+import { bookmarkLinks } from "@hoarder/db/schema";
+
+export async function GET(request: Request) {
+  const { searchParams } = new URL(request.url);
+  const assetIds = searchParams.getAll("assetId");
+
+  if (!assetIds || assetIds.length === 0) {
+    return new Response("", {
+      status: 404,
+    });
+  }
+
+  const bookmarkInformation = await db
+    .select({
+      title: bookmarkLinks.title,
+      htmlContent: bookmarkLinks.htmlContent,
+    })
+    .from(bookmarkLinks)
+    .where(inArray(bookmarkLinks.id, assetIds));
+
+  if (!bookmarkInformation || bookmarkInformation.length === 0) {
+    return new Response("", {
+      status: 404,
+    });
+  }
+
+  const chapters = bookmarkInformation.map((information) => {
+    return {
+      content: information.htmlContent ?? "",
+      title: information.title ?? "",
+    };
+  });
+
+  const title = getTitle(chapters);
+
+  const generatedEpub = await epub(title, chapters);
```

Comment:
This crashes if it fails to download some links, for example in this link (https://www.da.vidbuchanan.co.uk/blog/r1-jailbreak.html)

We might want to pass `ignoreFailedDownloads: true`.